### PR TITLE
Match openmct webpack config

### LINKED ISSUE #196: [Build] Update webpack config to match openmct
#### Summary
Open MCT has removed app.js and converted to a webpack-dev-server config. We should update this repo's implementation to match.

#### Current scope
- [ ] copy openmct's webpack.common.config or webpack.coverage.config
- [ ] Add proxy section
- [ ] Verify that proxy section works in CI and on local execution
- [ ] Update `npm start` to follow `openmct`'s script

### PR DESCRIPTION
**Pull request by scottbell**
Closes #196 

### Describe your changes:
Refactored webpack to match OpenMCT's setup.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Comment by unlikelyzero** [comment]
@scottbell can you add the coverage one and source it in the playwright e2e tests like we do in `nasa/openmct`?

File: package.json [hunk 1]
```diff
@@ -33,15 +33,16 @@
   "devDependencies": {
     "@babel/core": "7.19.6",
     "@babel/eslint-parser": "7.19.1",
-    "sinon": "14.0.0",
-    "uuid": "9.0.0",
+    "@playwright/test": "1.25.2",
     "eslint": "8.25.0",
     "eventemitter3": "4.0.7",
+    "sinon": "14.0.0",
     "source-map-loader": "^3.0.1",
```
Comment (unlikelyzero) [inline package.json:40]: @scottbell is this necessary anymore?

File: webpack.common.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+const WEBPACK_COMMON_CONFIG = {
+    performance: {
+        hints: false
+    },
+    module: {
+        rules: [
+            {
+                test: /\.js$/,
+                enforce: "pre",
+                use: ["source-map-loader"]
```
Comment (unlikelyzero) [inline webpack.common.js:35]: @scottbell Do you know why would be using source-map-loader on this project but not in `nasa/openmct`?

Comment (scottbell) [reply webpack.common.js:35]: If I had to guess, it's because we want to be able to have source-maps from the pulled-in code from `openmct`.

Comment (scottbell) [reply webpack.common.js:35]: Per the [library](https://webpack.js.org/loaders/source-map-loader/):
> This loader is especially useful when using 3rd-party libraries having their own source maps.

**Review by unlikelyzero — APPROVED** [review]

**Pull request merged**